pkg/oauthserver/oauthserver/auth.go

```diff
@@ -140,7 +140,7 @@ func (c *OAuthServerConfig) WithOAuth(handler http.Handler) (http.Handler, error
 	)
 	server.Install(mux, urls.OpenShiftOAuthAPIPrefix)
 
-	tokenRequestEndpoints := tokenrequest.NewTokenRequest(c.ExtraOAuthConfig.Options.MasterPublicURL, c.getOsinOAuthClient, c.ExtraOAuthConfig.OAuthAccessTokenClient)
+	tokenRequestEndpoints := tokenrequest.NewTokenRequest(c.ExtraOAuthConfig.Options.MasterPublicURL, c.getOsinOAuthClient)
```

Comment:
And then 5dddf472704af722175fb10e6226f3d6345c181e would not be trying to remove it.